sem/manager.py

```diff
@@ -349,36 +408,47 @@ def run_simulations(self, param_list, show_progress=True, stop_on_errors=True):
         else:
             result_generator = results
 
-        self.run_and_save_results(result_generator)
+        result_ids = self.run_and_save_results(result_generator, log_component)
 
-    def run_and_save_results(self, result_generator, batch_results=True):
+        if log_component:
```

Comment:
always check if not None. your implementation would return false if log_component is None, but also if log_component is an empty dictionary
```suggestion
        if log_component is not None:
```